dotcom-rendering/src/components/SlotBodyEnd.importable.tsx

```diff
@@ -200,7 +200,17 @@ export const SlotBodyEnd = ({
 			);
 	}, [isSignedIn, countryCode, brazeMessages, asyncArticleCount]);
 
-	if (SelectedEpic !== null) {
+	useEffect(() => {
+		if (SelectedEpic === null && showArticleEndSlot) {
+			document.dispatchEvent(
+				new CustomEvent('gu.commercial.slot.fill', {
+					detail: { slotId: 'dfp-ad--article-end' },
+				}),
+			);
+		}
+	});
```

Comment:
Without the dependency array this useEffect will run on every render. If we add a dependency array the function should only trigger when those specific dependencies change value - worth adding and the linter should tell you what needs to be added.

I think fill-advert-slots wont' overwrite an existing slot even if we do dispatch the vent multiple times but it is a little inefficient currently.

Reply:
Thanks! Just testing the theory on CODE atm